extensions/omni_vfs/migrate/4_table_fs.sql

```diff
@@ -0,0 +1,155 @@
+create table table_fs_filesystems
+(
+    id   integer primary key generated always as identity,
+    name text not null unique
+);
+
+create table table_fs_files
+(
+    id            bigint primary key generated always as identity,
+    filesystem_id integer                not null references table_fs_filesystems (id),
+    filename      text                 not null,
+    created_at    timestamp,
+    accessed_at   timestamp,
+    modified_at   timestamp,
+    -- disallow 'dir' entries because they are implicit
+    kind          omni_vfs_types_v1.file_kind not null check (kind != 'dir'),
+    unique (filesystem_id, filename)
+);
+
+create or replace function table_fs_files_trigger() returns trigger as
+$$
+begin
+    new.filename := omni_vfs.canonicalize_path(new.filename, absolute => true);
+    return new;
+end;
+$$ language plpgsql;
+
+create trigger table_fs_files_trigger
+    before insert or update
+    on table_fs_files
+    for each row
+execute function table_fs_files_trigger();
+
+
+-- TODO: refactor this into [sparse] [reusable] chunks
+create table table_fs_file_data
+(
+    file bigint primary key references table_fs_files (id),
+    /*  external storage optimizes fetching parts of data
+        at the expense of increased storage due to disabled compression
+    */
+    data bytea  storage external not null
```

Comment:
should there be a limit for data length ? perhaps something that can fit inside an `int` type because postgres builtin substring functions expects `int` type for offset parameter